common/src/main/java/com/wynntils/services/itemfilter/LevelSearchFilter.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter;
+
+import static java.lang.Integer.parseInt;
+
+import com.wynntils.models.items.WynnItem;
+import com.wynntils.models.items.properties.LeveledItemProperty;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class LevelSearchFilter extends ItemFilter {
+    private static final Pattern LEVEL_RANGE_PATTERN = Pattern.compile("^(\\d+)(-(\\d+))?$");
```

Comment:
If you make the optional part a non-matching group with `(?:` there will be no confusing group 2, but just group 1 and 2 (if present).
```
    private static final Pattern LEVEL_RANGE_PATTERN = Pattern.compile("^(\\d+)(?:-(\\d+))?$");
```